src/game/scripts/vscripts/abilities/pudge_flesh_heap_health_regeneration.lua

```diff
@@ -0,0 +1,143 @@
+
+
+function pudge_flesh_heap_health_regeneration:GetIntrinsicModifierName()
+  return "modifier_flesh_heap_health_regeneration"
+end
+
+modifier_flesh_heap_health_regeneration = class({})
+
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:IsHidden()
+    if self:GetAbility():GetLevel() == 0 then
+        return true
+    end
+    return false
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:RemoveOnDeath()
+    return false
+end
+
+--------------------------------------------------------------------------------
+function modifier_flesh_heap_health_regeneration:IsPurgable()
+    return false
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:IsPassive()
+    return true
+end
+
+function modifier_flesh_heap_health_regeneration:IsPurgable()
+    return true
+end
+
+function modifier_flesh_heap_health_regeneration:GetFleshHeapKills()
+  if self.nKills == nil then
+    self.nKills = 0
+  end
+  return self.nKills
+end
+ 
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:OnCreated( kv )
+  if not self:GetAbility() then
+    self:GetParent():RemoveModifierByName("modifier_flesh_heap_health_regeneration")
+    self:GetParent():CalculateStatBonus()
+    return
+  end
+  self.flesh_heap_value_buff_amount = self:GetAbility():GetSpecialValueFor( "flesh_heap_value_buff_amount" ) or 0
+  if IsServer() then
+    self:SetStackCount( self:GetFleshHeapKills() )
+    self:GetParent():CalculateStatBonus()
+  end
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:OnRefresh( kv )
+  if not self:GetAbility() then
+    self:GetParent():RemoveModifierByName("modifier_flesh_heap_health_regeneration")
+    self:GetParent():CalculateStatBonus()
+    return
+  end
+  self.flesh_heap_value_buff_amount = self:GetAbility():GetSpecialValueFor( "flesh_heap_value_buff_amount" ) or 0
+  if IsServer() then
+    self:SetStackCount( self:GetFleshHeapKills() )
+    self:GetParent():CalculateStatBonus()
+  end
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:DeclareFunctions()
+  local funcs = {
+    MODIFIER_PROPERTY_HEALTH_REGEN_PERCENTAGE,
+    MODIFIER_EVENT_ON_DEATH,
+  }
+  return funcs
+end
+
+function modifier_flesh_heap_health_regeneration:OnDeath(keys)
+
+
+  if not keys.unit or not keys.attacker then 
+    return 
+  end
+
+  if not keys.unit:IsRealHero() then
+    return 
+  end
+
+  if keys.unit:IsTempestDouble() then
+    return
+  end
+
+  if not IsServer() then 
```

Comment:
Yeah that one got pushed pack while it shouldnt have been, but those other functions are also extremely light and work clientside. I really regret not making a general function for this when I made the int and agi versions.